File: Java/프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        
        Map<String, Integer> types = new HashMap<>();
        types.put("R", 0);
        types.put("T", 0);
        types.put("C", 0);
        types.put("F", 0);
        types.put("J", 0);
        types.put("M", 0);
        types.put("A", 0);
        types.put("N", 0);
        
        for (int i = 0 ; i < survey.length ; i++) {
            String[] type = survey[i].split("");
            int choice = choices[i] - 4;
            
            if (choice > 0) {
                // type[1]
                types.put(type[1], types.get(type[1]) + choice);
            } else {
                // type[0]
                types.put(type[0], types.get(type[0]) - choice);
            }
        }
        
        String answer = "";
        if (types.get("R") >= types.get("T")) answer += "R";
        else answer += "T";
        
        if (types.get("C") >= types.get("F")) answer += "C";
        else answer += "F";
        
        if (types.get("J") >= types.get("M")) answer += "J";
        else answer += "M";
        
        if (types.get("A") >= types.get("N")) answer += "A";
        else answer += "N";
        
        return answer;
    }
}

File: JavaScript/프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    
    const type = {
        "RT": [0, 1],
        "TR": [0, -1],
        "CF": [1, 1],
        "FC": [1, -1],
        "JM": [2, 1],
        "MJ": [2, -1],
        "AN": [3, 1],
        "NA": [3, -1],
    };
    
    const types = [0, 0, 0, 0];
    for (let i = 0 ; i in survey ; i++) {
        const t = type[survey[i]];
        let choice = choices[i] - 4;
        types[t[0]] += choice * t[1];
    }
    
    let answer = '';
    if (types[0] > 0) {
        answer += "T";
    } else {
        answer += "R";
    }
    if (types[1] > 0) {
        answer += "F";
    } else {
        answer += "C";
    }
    if (types[2] > 0) {
        answer += "M";
    } else {
        answer += "J";
    }
    if (types[3] > 0) {
        answer += "N";
    } else {
        answer += "A";
    }
    
    return answer;
}